packages/snaps-utils/src/icon.ts

```diff
@@ -17,4 +22,41 @@ export const assertIsSnapIcon = (icon: VirtualFile) => {
   );
 
   assert(isSvg(icon.toString()), 'Snap icon must be a valid SVG.');
-};
+}
+
+/**
+ * Extract the dimensions of an image from an SVG string if possible.
+ *
+ * @param svg - An SVG string.
+ * @returns The height and width of the SVG or null.
+ */
+export function getSvgDimensions(svg: string): {
+  height: number;
+  width: number;
+} | null {
+  const parsed = parseSvg(svg);
+
+  const height = parsed['@_height'];
+  const width = parsed['@_width'];
+
+  if (height && width) {
+    return { height, width };
+  }
+
+  const viewBox = parsed['@_viewBox'];
```

Comment:
AFAIK the viewbox takes priority over width and height if it's set.

Reply:
Doesn't seem like that is the case.

If the viewBox differs from the height and width it functions as a zoom, the image is still output at width x height.